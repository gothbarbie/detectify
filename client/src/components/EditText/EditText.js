import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const TextInput = styled.input`
  background-color: transparent;
  border: none;
`

const EditText = ({ name, onChange, onBlur, value }) => (
  <TextInput name={name} onBlur={onBlur} onChange={onChange} value={value} />
)

EditText.propTypes = {
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
}

export default EditText
