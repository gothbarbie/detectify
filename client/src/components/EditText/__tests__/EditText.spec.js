import React from 'react'
import { shallow } from 'enzyme'

import EditText from '../EditText'

describe('Components/EditText', () => {
  it('renders', () => {
    const props = {
      onBlur: jest.fn(),
      onChange: jest.fn(),
      value: '',
    }
    const wrapper = shallow(<EditText {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
