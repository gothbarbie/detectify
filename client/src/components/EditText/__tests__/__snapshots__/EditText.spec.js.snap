// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/EditText renders 1`] = `
<styled.input
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  value=""
/>
`;
